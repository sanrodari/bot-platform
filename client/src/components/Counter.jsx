import React from 'react';
import PropTypes from 'prop-types';

export default function Counter({ count, increase, decrease }) {
  return (
    <div>
      <div>
        <span className="bold">Count:</span> {count}
      </div>
      <button onClick={increase} className="p1">
        Increase
      </button>
      <button onClick={decrease} className="p1">
        Decrease
      </button>
    </div>
  );
}

Counter.propTypes = {
  count: PropTypes.number.isRequired,
  increase: PropTypes.func.isRequired,
  decrease: PropTypes.func.isRequired,
};
