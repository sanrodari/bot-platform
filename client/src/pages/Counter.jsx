import React from 'react';
import { Helmet } from 'react-helmet';

import CounterComponent from '../components/Counter';
import counterContainer from '../containers/counter';

export default function Counter() {
  const EnhancedCounter = counterContainer(CounterComponent);

  return (
    <div>
      <Helmet>
        <title>Redux counter</title>
      </Helmet>
      <h1>Counter using redux</h1>

      <EnhancedCounter />
    </div>
  );
}
