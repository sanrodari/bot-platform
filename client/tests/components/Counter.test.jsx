import React from 'react';
import renderer from 'react-test-renderer';

import Counter from '../../src/components/Counter';

describe('<Counter />', () => {
  const props = {
    count: 0,
    increase: jest.fn(),
    decrease: jest.fn(),
  };

  it('renders correctly', () => {
    const component = renderer.create(<Counter {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
