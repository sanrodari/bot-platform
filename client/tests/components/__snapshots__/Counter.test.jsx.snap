// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders correctly 1`] = `
<div>
  <div>
    <span
      className="bold"
    >
      Count:
    </span>
     
    0
  </div>
  <button
    className="p1"
    onClick={[Function]}
  >
    Increase
  </button>
  <button
    className="p1"
    onClick={[Function]}
  >
    Decrease
  </button>
</div>
`;
